torch/csrc/jit/codegen/cuda/transform_iter.cpp

```diff
@@ -309,9 +330,13 @@ BestEffortReplay::BestEffortReplay(
     }
 
     // Check if any of the associated replay id's are part of an rfactor domain
-    bool replay_has_rfactor_inp =
-        std::any_of(replay_inps.begin(), replay_inps.end(), [](IterDomain* id) {
-          return id == nullptr ? false : id->isRFactorProduct();
+    bool replay_has_rfactor_inp = std::any_of(
+        replay_inps.begin(),
+        replay_inps.end(),
+        [&replay_rfactor_ids](IterDomain* id) {
+          return id == nullptr ? false
+                               : id->isRFactorProduct() &&
```

Comment:
Is `id->isRFactorProduct() &&` necessary?

Reply:
No, this check shouldn't be necessary anymore.